### DISCUSSION TITLE
Expose input cache hook to track value after unregister

### DISCUSSION THREAD

**Discussion by lfarrel6** [react-hook-form/react-hook-form#3470] [Ideas] [answered]
I was implementing a form which required an input to show/hide depending on the state of another input (in this case a checkbox). Looking at the docs, I read that the form doesn't track inputs after they unregister. 

The two sets of behaviours from `shouldUnregister` seemed as though they would add significant boilerplate with the ideal behaviour being some combination of the two - persist the value, and not have to update the form state.

I built a hook to fit my use case (it's pretty naive in how it detects the mount/unmount events) which seems to allow this behaviour - [here's a simple example of it.](https://codesandbox.io/s/headless-sound-wgetb)

Would it make sense to include this hook as part of the library? Would love to hear your thoughts.

Awesome library by the way, thank you for all your hard work!

**Comment by bluebill1049** [comment, marked as answer]
hey @lfarrel6 sorry about the late reply, I have saved in my bookmark and forgot to reply. your custom hook is pretty smart! I think let's keep it at your end for now, but would you like to send a PR with this as an example (our /examples folder).

**Reply by lfarrel6** [reply]
No worries! 

Sure thing, I'll PR it when I get a chance later in the week

**Reply by lfarrel6** [reply]
I've PR'd a simple example of the hook [here](https://github.com/react-hook-form/react-hook-form/pull/3745)

Lmk if there's any changes to be made on the PR!

**Reply by bluebill1049** [reply]
merged :)
